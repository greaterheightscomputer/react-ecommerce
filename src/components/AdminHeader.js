import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const AdminHeader = ({ startLogout }) => (
    <header className="admin-header">        
        <div className="content-container">
            <div className="admin-header__container">
                <Link className="admin-header__title " to="/admin_dashboard">
                    <h1>E-Market</h1>
                </Link>
                <button className="button button--link" onClick={startLogout}>Logout</button>          
            </div>            
        </div>        
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(AdminHeader);