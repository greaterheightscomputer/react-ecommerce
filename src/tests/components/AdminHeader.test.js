import React from 'react';
import { shallow } from 'enzyme';
import AdminHeader from '../../components/AdminHeader';

test('should render AdminHeader correctly', () => {
    const wrapper = shallow(<AdminHeader />);
    expect(wrapper).toMatchSnapshot();
});