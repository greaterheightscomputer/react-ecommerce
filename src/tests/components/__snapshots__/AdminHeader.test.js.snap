// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AdminHeader correctly 1`] = `
<header>
  <h1>
    OyinMark
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Customer
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/admin_dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create Product
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/edit"
  >
    Edit Product
  </NavLink>
</header>
`;
